pyvene/models/intervenable_base.py

```diff
@@ -717,18 +710,11 @@ def _intervention_getter(
         Create a list of getter handlers that will fetch activations
         """
         handlers = []
-        for key_i, key in enumerate(keys):
+        for key in keys:
+            key_i = self.sorted_keys.index(key)
             intervention, module_hook = self.interventions[key]
 
             def hook_callback(model, args, kwargs, output=None):
-                if self._is_generation:
```

Comment:
This was dead code already iirc, since it's just getting passed here. Correct me if I'm wrong, but since the getter hook is used to gather source representations wouldn't it still need to run even if a `generate()` call skips intervening on the base (prompt)?